example.c

```diff
@@ -21,9 +21,9 @@
 #include <stdio.h>
 #include <string.h>
 
-static void read_data(char *buf, size_t len)
```

Comment:
In my opinion it seems this commit is actually 3-4 commits put together. I would separate different actions into their own commit.